pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1351,6 +1351,15 @@ func (n Nodes) Find(f func(NodeStatus) bool) *NodeStatus {
 	return nil
 }
 
+func (n Nodes) HasHTTPNodes() bool {
```

Comment:
Minor - maybe call this “HasTaskSetNode” so we can add other types in the future